src/hooks/useOnEscape.ts

```diff
@@ -0,0 +1,21 @@
+import { useCallback, useEffect } from "react";
+
+/**
+ * Handle the escape key pressed.
+ */
+export const useOnEscape = (onEscape: () => void) => {
```

Comment:
Might be better to be more explicit here.
```suggestion
export const useOnEscapeKeyDown = (onEscape: () => void) => {
```
```suggestion
export const useOnEscapeKey = (onEscape: () => void) => {
```